test/cfg/qt.cpp

```diff
@@ -39,6 +40,29 @@ class MacroTest1: public QObject {
     ~MacroTest1();
 };
 
+class MacroTest2 {
+    Q_DECLARE_TR_FUNCTIONS(MacroTest2)
+
+public:
+    MacroTest2();
+    ~MacroTest2();
+};
+
+void MacroTest2_test()
+{
+    // cppcheck-suppress checkLibraryFunction
+    QString str = MacroTest2::tr("hello");
```

Comment:
I wonder why i see this `checkLibraryFunction` message here.
The declaration of `MacroTest2` with the full inline definition of `tr()` is directly above.
@danmar Is that something you would expect or could it be a problem with the macro definition in the library configuration?

Reply:
This is the corresponding code in qt.cpp after preprocessed by Cppcheck:
```
class MacroTest2 {
 public  :  static  inline  QString  tr  (  const  char  *  sourceText  ,  const  char  *  disambiguation  =  NULL  ,  int  n  =  -  1  )  {  return  QCoreApplication  ::  translate  ( "MacroTest2"  ,  sourceText  ,  disambiguation  ,  n  )  ;  }  static  inline  QString  trUtf8  (  const  char  *  sourceText  ,  const  char  *  disambiguation  =  NULL  ,  int  n  =  -  1  )  {  return  QCoreApplication  ::  translate  (  "MacroTest2"  ,  sourceText  ,  disambiguation  ,  n  )  ;  }  private  :

public :
MacroTest2 ( ) ;
~ MacroTest2 ( ) ;
} ;

void MacroTest2_test ( )
{

QString str = MacroTest2 :: tr ( "hello" ) ;
QByteArray ba = str . toLatin1 ( ) ;
printf ( ba . data ( ) ) ;



str = MacroTest2 :: trUtf8 ( "test2" ) ;
ba = str . toLatin1 ( ) ;
printf ( ba . data ( ) ) ;

}
```
g++ has no problem with it, so i guess the macros are valid.

Reply:
not sure if I understand correctly. Do you think it's some preprocessor issue, or a `checkLibraryFunction` issue (it should not be reported because it is defined, right?)

Reply:
The preprocessor output looks good. I do not see a problem there.
Yes, it should not be reported as function missing a configuration.
I will see if I can create a reduced example.